flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnConfigurationITCase.java

```diff
@@ -208,4 +215,12 @@ private boolean hasTaskManagerConnectedAndReportedSlots(Collection<TaskManagerIn
 			return taskManagerInfo.getNumberSlots() > 0;
 		}
 	}
+
+	private static int calculateManagedMemorySizeMB(Configuration originalConfiguration, int numSlotsPerTaskManager) {
+		Configuration configuration = new Configuration(originalConfiguration); // copy, because we alter the config
+
+		final int defaultTaskManagerMemoryMB = ConfigurationUtils.getTaskManagerHeapMemory(configuration).getMebiBytes();
+		YarnResourceManager.updateTaskManagerConfigAndCreateWorkerSlotProfiles(configuration, defaultTaskManagerMemoryMB, numSlotsPerTaskManager);
```

Comment:
I would suggest to let such a method return the modified configuration instead of using an output parameter.